go/pkg/sdk/sdk_test.go

```diff
@@ -259,3 +404,126 @@ func TestSdkGetSupportedStablecoins(t *testing.T) {
 		})
 	}
 }
+
+func printUserBalance(from common.Address, client *ethclient.Client) {
+	balance, err := client.BalanceAt(context.Background(), from, nil)
+	if err != nil {
+		log.Fatalf("Failed to get the balance: %v", err)
+	}
+	ethBalance := new(big.Float).Quo(new(big.Float).SetInt(balance), big.NewFloat(1e18))
+	fmt.Printf("Eth balance of %s: %f ETH\n", from.Hex(), ethBalance)
+}
+
+func getAssetUserBalance(from common.Address, asset common.Address, client *ethclient.Client, chainId int64, isDeposit bool) (*big.Int, error) {
+	var balanceAsset *big.Int
+
+	if chainId == 1 {
+		if isDeposit {
+			assetContract, err := ethereumContracts.NewDaiCaller(asset, client)
```

Comment:
note: it's a shame that we can't have a generic interface to get the balance of an asset. I'm not asking for changes, I just wanted to share a point of view

Reply:
instead of `if isDeposit` we should have something like:

```go
if cond1 {
    // Do something if cond1 is true
} else if cond2 {
    // Do something else if cond2 is true
} else {
    return fmt.Errorf("An error occurred")
}
```